src/Symfony/Component/Security/Http/Tests/EventListener/PasswordMigratingListenerTest.php

```diff
@@ -91,3 +103,22 @@ private function createEvent(PassportInterface $passport)
         return new LoginSuccessEvent($this->createMock(AuthenticatorInterface::class), $passport, $this->createMock(TokenInterface::class), new Request(), null, 'main');
     }
 }
+
+class MigratingUserProvider implements UserProviderInterface, PasswordUpgraderInterface
+{
+    public function upgradePassword(UserInterface $user, string $newEncodedPassword): void
+    {
+    }
+
+    public function loadUserByUsername(string $username)
+    {
+    }
+
+    public function refreshUser(UserInterface $user)
+    {
+    }
+
+    public function supportsClass(string $class)
+    {
+    }
+}
```

Comment:
```suggestion
abstract class MigratingUserProvider implements UserProviderInterface, PasswordUpgraderInterface
{
}
```

Because you only need this class for mocking, you can just set it abstract and remove the empty method bodies.

Reply:
Thank you!